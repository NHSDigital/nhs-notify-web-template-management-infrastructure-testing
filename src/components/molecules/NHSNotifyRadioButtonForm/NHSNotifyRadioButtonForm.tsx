import { Radios, Button, Fieldset } from 'nhsuk-react-components';
import { FormState } from '../../../utils/types';

export type NHSNotifyRadioButtonFormProps = {
    radiosId: string;
    action: string | ((payload: FormData) => void);
    state: FormState;
    pageHeading: string;
    options: {
        id: string;
        text: string;
    }[];
    buttonText: string;
};

export const NHSNotifyRadioButtonForm = ({
    radiosId,
    action,
    state,
    pageHeading,
    options,
    buttonText,
}: NHSNotifyRadioButtonFormProps) => (
    <form action={action}>
        <Fieldset>
            <Fieldset.Legend isPageHeading={true} size='l'>
                {pageHeading}
            </Fieldset.Legend>
            <Radios id={radiosId} error={state.fieldErrors[radiosId]?.join(', ')} errorProps={{ id: `${radiosId}-error-message`, }}>
                {
                    options.map(({ id, text }) => (
                        <Radios.Radio value={id} data-testid={`${id}-radio`} key={`${id}-radio`}>
                            {text}
                        </Radios.Radio>
                    ))
                }
            </Radios>
        </Fieldset>
        <Button type='submit' data-testid='submit-button'>{buttonText}</Button>
    </form>
);
