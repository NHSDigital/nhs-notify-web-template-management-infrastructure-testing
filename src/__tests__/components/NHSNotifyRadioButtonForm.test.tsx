import { render } from '@testing-library/react';
import { NHSNotifyRadioButtonForm } from '../../components/molecules/NHSNotifyRadioButtonForm/NHSNotifyRadioButtonForm';

test('Renders NHSNotifyRadioButtonForm correctly without errors', () => {
  const container = render(
    <NHSNotifyRadioButtonForm
      radiosId='radios-id'
      action='/action'
      state={{
        fieldErrors: {},
        formErrors: [],
      }}
      pageHeading={'Page heading'}
      options={[
        { id: 'option-1', text: 'option 1' },
        { id: 'option-2', text: 'option 2' },
      ]}
      buttonText='Continue'
    />
  );

  expect(container.asFragment()).toMatchSnapshot();
});

test('Renders NHSNotifyRadioButtonForm correctly with errors', () => {
  const container = render(
    <NHSNotifyRadioButtonForm
      radiosId='radios-id'
      action='/action'
      state={{
        fieldErrors: {
          'radios-id': ['Field error'],
        },
        formErrors: ['Form error'],
      }}
      pageHeading={'Page heading'}
      options={[
        { id: 'option-1', text: 'option 1' },
        { id: 'option-2', text: 'option 2' },
      ]}
      buttonText='Continue'
    />
  );

  expect(container.asFragment()).toMatchSnapshot();
});
