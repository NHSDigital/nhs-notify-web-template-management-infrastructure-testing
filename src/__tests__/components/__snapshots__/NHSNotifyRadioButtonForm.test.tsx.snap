// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders NHSNotifyRadioButtonForm correctly with errors 1`] = `
<DocumentFragment>
  <form
    action="/action"
  >
    <div
      class="nhsuk-form-group nhsuk-form-group--error"
    >
      <fieldset
        class="nhsuk-fieldset"
      >
        <legend
          class="nhsuk-fieldset__legend nhsuk-fieldset__legend--l"
        >
          <h1
            class="nhsuk-fieldset__heading"
          >
            Page heading
          </h1>
        </legend>
        <div
          class="nhsuk-form-group nhsuk-form-group--error"
        >
          <span
            class="nhsuk-error-message"
            id="radios-id-error-message"
          >
            <span
              class="nhsuk-u-visually-hidden"
            >
              Error: 
            </span>
            Field error
          </span>
          <div
            class="nhsuk-radios"
            id="radios-id"
          >
            <div
              class="nhsuk-radios__item"
            >
              <input
                aria-labelledby="radios-id-1--label"
                class="nhsuk-radios__input"
                data-testid="option-1-radio"
                id="radios-id-1"
                name="radios-id"
                type="radio"
                value="option-1"
              />
              <label
                class="nhsuk-label nhsuk-radios__label"
                for="radios-id-1"
                id="radios-id-1--label"
              >
                option 1
              </label>
            </div>
            <div
              class="nhsuk-radios__item"
            >
              <input
                aria-labelledby="radios-id-2--label"
                class="nhsuk-radios__input"
                data-testid="option-2-radio"
                id="radios-id-2"
                name="radios-id"
                type="radio"
                value="option-2"
              />
              <label
                class="nhsuk-label nhsuk-radios__label"
                for="radios-id-2"
                id="radios-id-2--label"
              >
                option 2
              </label>
            </div>
          </div>
        </div>
      </fieldset>
    </div>
    <button
      aria-disabled="false"
      class="nhsuk-button"
      data-testid="submit-button"
      type="submit"
    >
      Continue
    </button>
  </form>
</DocumentFragment>
`;

exports[`Renders NHSNotifyRadioButtonForm correctly without errors 1`] = `
<DocumentFragment>
  <form
    action="/action"
  >
    <div
      class="nhsuk-form-group"
    >
      <fieldset
        class="nhsuk-fieldset"
      >
        <legend
          class="nhsuk-fieldset__legend nhsuk-fieldset__legend--l"
        >
          <h1
            class="nhsuk-fieldset__heading"
          >
            Page heading
          </h1>
        </legend>
        <div
          class="nhsuk-form-group"
        >
          <div
            class="nhsuk-radios"
            id="radios-id"
          >
            <div
              class="nhsuk-radios__item"
            >
              <input
                aria-labelledby="radios-id-1--label"
                class="nhsuk-radios__input"
                data-testid="option-1-radio"
                id="radios-id-1"
                name="radios-id"
                type="radio"
                value="option-1"
              />
              <label
                class="nhsuk-label nhsuk-radios__label"
                for="radios-id-1"
                id="radios-id-1--label"
              >
                option 1
              </label>
            </div>
            <div
              class="nhsuk-radios__item"
            >
              <input
                aria-labelledby="radios-id-2--label"
                class="nhsuk-radios__input"
                data-testid="option-2-radio"
                id="radios-id-2"
                name="radios-id"
                type="radio"
                value="option-2"
              />
              <label
                class="nhsuk-label nhsuk-radios__label"
                for="radios-id-2"
                id="radios-id-2--label"
              >
                option 2
              </label>
            </div>
          </div>
        </div>
      </fieldset>
    </div>
    <button
      aria-disabled="false"
      class="nhsuk-button"
      data-testid="submit-button"
      type="submit"
    >
      Continue
    </button>
  </form>
</DocumentFragment>
`;
